### PR TITLE
Engine should support health check per context

### PR DESCRIPTION
**Pull request by phuongnq**
### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/6020
https://github.com/craftercms/craftercms/issues/6021

### CODE REVIEW

**Comment by phuongnq** [comment]
@avasquez614 The `site` parameter is a part of the requirement here: https://github.com/craftercms/craftercms/issues/6021
It's optional. If you don't send the `site` parameter, it works as you mention. See https://github.com/craftercms/engine/pull/623/files#diff-0e0bb077eba43bc03befaf4b04e43a0ae0562df7f96b1ddc182a1c334ea1d907R90

**Review by avasquez614 — APPROVED** [review]
Understood, sorry I hadn't seen the code in the bottom :D

File: src/main/java/org/craftercms/engine/exception/SiteContextInvalidException.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.craftercms.engine.exception;
```
Comment (jmendeza) [inline src/main/java/org/craftercms/engine/exception/SiteContextInvalidException.java:1]: Missing copyright notice?

File: src/main/java/org/craftercms/engine/controller/rest/MonitoringController.java [hunk 2]
```diff
@@ -56,4 +69,29 @@ public List<Map<String, Object>> getLoggedEvents(@RequestParam @ValidSiteId Stri
...
+                                       @RequestParam(name = "token") String token)
+            throws InvalidManagementTokenException {
+        validateToken(token);
+
+        if (site != null) {
+            String paramNameValidationKey = SITE_ID.typeKey;
+            try {
+                validator.getValidInput(paramNameValidationKey, site, paramNameValidationKey, MAXIMUM_SITE_ID_KEY_LENGTH, false);
+            } catch (ValidationException e) {
+                throw new SiteContextInvalidException(format("Invalid site Id: '%s'.", site));
```
Comment (jmendeza) [inline src/main/java/org/craftercms/engine/controller/rest/MonitoringController.java:84]: Will this return a 500?
I would expect a 400 in the case the actual site ID is an invalid value.

cc @sumerjabri @avasquez614

**Review by jmendeza — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/engine/exception/SiteContextInvalidException.java (hunk 1 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/engine/exception/SiteContextInvalidException.java:16]: Updated. Thanks!

File: src/main/java/org/craftercms/engine/controller/rest/MonitoringController.java (hunk 2 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/engine/controller/rest/MonitoringController.java]: Updated to return 400 if the actual site ID is an invalid one. Thanks!

**Review by jmendeza — APPROVED** [review]

**Pull request merged**